clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -307,6 +285,36 @@ public HubConnectionState getConnectionState() {
         return stop(null);
     }
 
+    private void stopConnection(String errorMessage) {
+        HubException hubException = null;
+        hubConnectionStateLock.lock();
+        try {
+            hubConnectionState = HubConnectionState.DISCONNECTED;
+
+            // errorMessage gets passed in from the transport. An already existing stopError value
+            // should take precedence.
+            if (stopError != null) {
```

Comment:
set `stopError` to null somewhere so if they reuse the connection and close it gracefully they don't get the error string